### PR TITLE
Feature/v2 updates

### PR DESCRIPTION
**Pull request by fivetran-reneeli**
Pull Request
**Are you a current Fivetran customer?** 

Internal
**What change(s) does this PR introduce?** 

- Removal of the `snapchat__ad_adapter` model
- Update the `account`, `ad_squad`, and `campaign` models
- Addition of `ad`, `creative`, `UTM` report models
- Addition of testing for said models to ensure granularity and accuracy of the modeled data.
- Update package to meet new standardization criteria
- 
**Did you update the CHANGELOG?** 

- [x] Yes

**Does this PR introduce a breaking change?**

- [x] Yes (please provide breaking change details below.)
- [ ] No  (please provide an explanation as to how the change is non-breaking below.)

**Did you update the dbt_project.yml files with the version upgrade (please leverage standard semantic versioning)? (In both your main project and integration_tests)** 

- [x] Yes

**Is this PR in response to a previously created Bug or Feature Request**

- [ ] Yes, Issue/Feature [link bug/feature number here]
- [x] No 

**How did you test the PR changes?** 

 

- [x] CircleCi  
- [x] Local (please provide additional testing details below)

**Select which warehouse(s) were used to test the PR**
 

- [x] BigQuery
- [x] Redshift
- [x] Snowflake
- [x] Postgres
- [x] Databricks
- [ ] Other (provide details below)

**Provide an emoji that best describes your current mood**
 
:dancer:

**Feedback**

We are so excited you decided to contribute to the Fivetran community dbt package! We continue to work to improve the packages and would greatly appreciate your [feedback](https://www.surveymonkey.com/r/DQ7K7WW) on our existing dbt packages or what you'd like to see next.

### CODE REVIEW

File: models/snapchat__utm_report.sql
Comment (fivetran-joemarkiewicz) [inline models/snapchat__utm_report.sql:76]: We decided that the url report should contain _only_ ads that utilize utm parameters. Can you follow the same modeling logic used in Google Ads to filter out the non url ads. All ads are already captured in the ad_report.

https://github.com/fivetran/dbt_google_ads/blob/8c2240828ed58067d5ecd32203652cf606bf446a/models/google_ads__url_report.sql#L86-L88

File: packages.yml
Comment (fivetran-joemarkiewicz) [inline packages.yml:7]: Friendly reminder to switch before merge

File: README.md
Comment (fivetran-joemarkiewicz) [inline README.md:18]:
```suggestion
- Produces modeled tables that leverage Snapchat Ads data from [Fivetran's connector](https://fivetran.com/docs/applications/snapchat-ads) in the format described by [this ERD](https://fivetran.com/docs/applications/snapchat-ads#schemainformation) and builds off the output of our [Snapchat Ads source package](https://github.com/fivetran/dbt_snapchat_ads_source).
```

File: README.md
Comment (fivetran-joemarkiewicz) [inline README.md:25]: Can you add the dbt docs links to each of the models provided in this table.

File: README.md
Comment (fivetran-joemarkiewicz) [inline README.md:30]: We want to change this to be url report instead of utm

File: README.md
Comment (fivetran-joemarkiewicz) [inline README.md:65]: These variables in the source are actually `snapchat_schema` and `snapchat_database`

File: README.md
Comment (fivetran-joemarkiewicz) [inline README.md:78]:
```suggestion
```

We should remove until the dbt hub render issue is resolved.

File: README.md
Comment (fivetran-joemarkiewicz) [inline README.md:91]:
```suggestion
```

We should remove until the dbt hub render issue is resolved.

File: README.md
Comment (fivetran-joemarkiewicz) [inline README.md:95]:
```suggestion
```

We should remove until the dbt hub render issue is resolved.

File: README.md
Comment (fivetran-joemarkiewicz) [inline README.md:98]:
```suggestion
```

We should remove until the dbt hub render issue is resolved.

File: README.md
Comment (fivetran-joemarkiewicz) [inline README.md:115]: We should add a reference to the DECISIONLOG.md in this section. See how we did it [here](https://github.com/fivetran/dbt_google_ads/tree/feature/v2_updates#opinionated-decisions)

File: DECISIONLOG.md
Comment (fivetran-joemarkiewicz) [inline DECISIONLOG.md:1]: We should also add the UTM filtering details here as well. See the decisionlog entry for [Google Ads](https://github.com/fivetran/dbt_google_ads/blob/feature/v2_updates/DECISIONLOG.md#utm-report-filtering) for format and reasoning.

> The UTM filtering comment is in a comment attached to the utm report model.

File: models/snapchat.yml
Comment (fivetran-joemarkiewicz) [inline models/snapchat.yml:119]: We should specify here that this model is just for the url ads.

File: models/snapchat.yml
Comment (fivetran-joemarkiewicz) [inline models/snapchat.yml:30]: This is the same description as the `snapchat__account_report` model. Can we add the proper description for this model.

File: models/snapchat__campaign_report.sql
Comment (fivetran-joemarkiewicz) [inline models/snapchat__campaign_report.sql:16]: Do we need to to bring ad_squads into this model? It doesn't look like we use it in the final cte and the join isn't required to get the ad account info.

File: models/snapchat__utm_report.sql
Comment (fivetran-joemarkiewicz) [inline models/snapchat__utm_report.sql:1]: I see in this model we are attributing campaigns and ad_squads to the utm ads. I thought this could cause duplicate results? Should we be removing those fields from this model as well as the ad report?

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
@fivetran-reneeli Thanks for this PR! I have a few requested updates to be applied before this PR will be good to go. Let me know if you have any questions.
- [ ] We recently renamed the utm report to be the url report. We will want to update the model name and the yml documented reference.
- [ ] The staging models are all named `snapchat_ads__*`; however, this package models are named `snapchat__*`. Can we update these models and the yml documentation to be consistent.
- [ ] It looks like we include the campaign and ad squad data in the urtm (to be renamed url) model. Is there a reason we should include them in that model and not the ad report? I think it would still cause duplicates. Right?
- [ ] The `platform` field will be added in the ad reporting models. We can remove them from the package models.
- [ ] The currency field seems to be missing from the campaign report. Can you add that in before regenerating the docs.
- [ ] Regenerate and add the docs.
- [ ] Apply a filter to the url report to filter out fields that do not contain utm parameters.
- [ ] Same comment from the source regarding if we should add passthrough metrics to the end models.
- [ ] Address any other inline comments within this review. Remember to expand the comments if GitHub collapses them.

File: CHANGELOG.md
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:1]: Same comment from the source. Can we add the model renaming (both renaming from the source and transform packages but especially the transform renames) and the variable name changes from the source package as breaking changes to this entry as well.

File: README.md
Comment (fivetran-joemarkiewicz) [inline README.md:78]: Can we also add in the [change the source table references](https://github.com/fivetran/dbt_snapchat_ads_source/tree/feature/v2_updates#change-the-source-table-references) section to this README.

File: models/snapchat_ads__account_report.sql
Comment (fivetran-joemarkiewicz) [inline models/snapchat_ads__account_report.sql:40]: `roung()` without the precision defined will round to the nearest dollar. For example:
```txt
round(1.4445) = 1.0
```
To be safe, can we add a precision of 2 in this round statement. I don't want us to round down or up to the dollar. The 100th place seems comfortable though.

Comment (fivetran-joemarkiewicz) [reply models/snapchat_ads__account_report.sql:40]: Can you make this update to all other models in this transform package.

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
@fivetran-reneeli thanks for applying the requested changes! The PR is looking great and is almost there. I have just a few small updates that should be made to the package before approval. In addition to inline comments, a few updates to highlight include:
- [ ] Minor README updates
- [x] CHANGELOG updates to highlight the breaking changes (especially model and variable renaming).
- [x] Update the `round()` functions to include double precision (eg. `round(field,2)`)
- [x] Regen docs.

File: models/snapchat_ads__account_report.sql
Comment (fivetran-reneeli) [reply models/snapchat_ads__account_report.sql]: Oh good good point, it definitely should be in the cents, not whole dollar!

File: CHANGELOG.md
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:20]: Small request to move this to the top of the release notes. This way the breaking changes are the first thing people see before upgrading.

**Review by fivetran-joemarkiewicz — APPROVED** [review]
@fivetran-reneeli thanks so much for applying these changes! This looks good to go (friendly reminder that we will hold off merging until all other packages are ready).

Only request I have is to adjust the CHANGELOG to put the breaking changes at the top of the release notes.

**Pull request merged**
